### PR TITLE
Add resolutionPolicy property to VirtualNetworkLink resource in privatedns

### PR DESCRIPTION
**Pull request by pranavpnair**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

# PR Description

A new (enum) property called NxDomainRedirect is being added to VirtualNetworkLink resource. 
The property was already added to swagger in July. The PR can be found here: https://github.com/Azure/azure-rest-api-specs/pull/29470

The default API version for PrivateDNS functions has also been updated to the latest available version: 2024-06-01.

Change owners: prannair@microsoft.com, rdnet-privatedns-dev@microsoft.com

### CODE REVIEW

**Comment by pranavpnair** [comment]
A new (enum) property called NxDomainRedirect is being added to VirtualNetworkLink resource. 
The property was already added to swagger in July. The PR can be found here: https://github.com/Azure/azure-rest-api-specs/pull/29470

**Review by rahisshaikh — COMMENTED** [review]
PrivateDNS team has reviewed and signed off on the change

File: sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs [hunk 1]
```diff
@@ -174,6 +189,47 @@ public async Task AddRemoveTag(bool? useTagResource)
...
+            var virtualNetworkLinkData = new VirtualNetworkLinkData("global")
+            {
+                VirtualNetwork = new WritableSubResource { Id = vnet.Id },
+                RegistrationEnabled = false,
+                ResolutionPolicy = ResolutionPolicy.NxDomainRedirect,
+            };
+
+            await CreateOrUpdateVirtualNetworkLink(privateZone, vnetLinkName, virtualNetworkLinkData);
+
+            if (TestEnvironment.Mode == RecordedTestMode.Record)
```
Comment (archerzz) [inline sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs:211]: The above operation is a long running operation, and you invoke it with `WaitUntil.Completed`. So, here we should not wait another 30 seconds, right?

**Review by archerzz — APPROVED** [review]
Generally good. Just one minor comment regarding the test case.

File: sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs [hunk 2]
```diff
@@ -174,6 +189,47 @@ public async Task AddRemoveTag(bool? useTagResource)
...
+            }
+
+            var virtualNetworkLinkResource = await privateZone.GetVirtualNetworkLinks().GetAsync(vnetLinkName);
+            var vnetLinkData = virtualNetworkLinkResource.Value.Data;
+            Assert.AreEqual(vnetLinkData.ResolutionPolicy, virtualNetworkLinkData.ResolutionPolicy);
+
+            // Update resolutionPolicy to default
+            virtualNetworkLinkData.ResolutionPolicy = ResolutionPolicy.Default;
+            await CreateOrUpdateVirtualNetworkLink(privateZone, vnetLinkName, virtualNetworkLinkData);
+            if (TestEnvironment.Mode == RecordedTestMode.Record)
```
Comment (yashlaxmigupta) [inline sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs:223]: Just curious what is the behaviour for "record" here. I think the other enum values are "live" and "playback".

File: sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs (hunk 1 above)
Comment (pranavpnair) [reply sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs]: Yes you're right, that was unnecessary. Removed it in the latest commit.

File: sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs (hunk 2 above)
Comment (pranavpnair) [reply sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs]: Record is similar to Live but it records the requests/responses in a json file (which need to be uploaded to a different repo), which is then used for Playback mode. Playback mode does not make actual network calls, and uses the json to simulate request/response.

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**